test/preview/dataclasses/test_document.py

```diff
@@ -1,72 +1,80 @@
+    doc2 = Document.from_dict(doc2_data)
+    assert doc1.id == doc2.id
 
 
 @pytest.mark.unit
-def test_content_type_must_match_text():
-    Document(content="test content")
-    Document(content="test content", content_type="text")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content="test content", content_type="table")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content="test content", content_type="image")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content="test content", content_type="audio")
+@pytest.mark.parametrize(
+    "doc1_data,doc2_data",
+    [
+        [{"text": "test text"}, {"text": "test text "}],
+        [{"array": np.array(range(10))}, {"array": np.array(range(11))}],
+        [{"dataframe": pd.DataFrame([1, 2, 3])}, {"dataframe": pd.DataFrame([1, 2, 3, 4])}],
+        [{"blob": b"some bytes"}, {"blob": "something else".encode()}],
+    ],
+)
+def test_id_hash_keys_default_fields_different_ids(doc1_data, doc2_data):
+    doc1 = Document.from_dict(doc1_data)
+    doc2 = Document.from_dict(doc2_data)
+    assert doc1.id != doc2.id
 
 
 @pytest.mark.unit
-def test_content_type_must_match_table():
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content=pd.DataFrame([1, 2]), content_type="text")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content=pd.DataFrame([1, 2]), content_type="image")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content=pd.DataFrame([1, 2]), content_type="audio")
-
-
-@pytest.mark.unit
-def test_content_type_must_match_image_audio():
-    # Mimetypes are not checked - yet
-    Document(content=Path(__file__), content_type="image")
-    Document(content=Path(__file__), content_type="audio")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content=Path(__file__), content_type="text")
-
-    with pytest.raises(ValueError, match="does not match the content type"):
-        Document(content=Path(__file__), content_type="table")
+def test_id_hash_keys_changes_id():
+    doc1 = Document(text="test text", metadata={"some-value": "value"})
+    doc2 = Document(text="test text", metadata={"some-value": "value"}, id_hash_keys=["text", "some-value"])
+    assert doc1.id != doc2.id
 
 
 @pytest.mark.unit
-def test_id_hash_keys_require_metadata():
-    with pytest.raises(ValueError, match="must be present in the metadata of the Document if you want to use it"):
-        Document(content="test content", id_hash_keys=["something"])
+def test_id_hash_keys_field_may_be_missing():
+    doc1 = Document(text="test text", id_hash_keys=["something"])
+    doc2 = Document(text="test text", id_hash_keys=["something else"])
+    assert doc1.id == doc2.id
```

Comment:
Maybe we can add a check here that we are issuing a log message about the missing field.